packages/adena-extension/src/components/transfer/transfer-summary/transfer-summary.tsx

```diff
@@ -18,59 +20,65 @@ export interface TransferSummaryProps {
     value: string;
     denom: string;
   };
+  transfer: () => Promise<boolean>;
+  onClickLedgerCancel: () => void;
   onClickBack: () => void;
   onClickCancel: () => void;
   onClickSend: () => void;
 }
 
 const TransferSummary: React.FC<TransferSummaryProps> = ({
+  isLoadingLedger,
   tokenMetainfo,
   tokenImage,
   transferBalance,
   toAddress,
   networkFee,
+  transfer,
+  onClickLedgerCancel,
   onClickBack,
   onClickCancel,
   onClickSend,
 }) => {
-  return (
-    <TransferSummaryWrapper>
-      <div className='sub-header-wrapper'>
-        <SubHeader
-          leftElement={{
-            element: <img src={`${ArrowLeftIcon}`} alt={'back image'} />,
-            onClick: onClickBack
-          }}
-          title={`Sending ${tokenMetainfo.symbol}`}
-        />
-      </div>
-      <div className='info-wrapper'>
-        <TransferSummaryBalance
-          tokenImage={tokenImage}
-          value={transferBalance.value}
-          denom={transferBalance.denom}
-        />
-
-        <div className='direction-icon-wrapper'>
-          <img src={`${ArrowDownIcon}`} alt='direction-icon' />
+  return isLoadingLedger ?
```

Comment:
I think we should just return it here.
```
if (isLoadingLedger) {
  return <SendConfirmLedgerLoading transfer={transfer} onClickLedgerCancel={onClickLedgerCancel} />;
}
```

Reply:
This was solved by creating a separate ledger loading page and reject page.